### ISSUE TITLE
WaveShaperNode connected to OfflineAudioContext doesn't render last sample

### ISSUE THREAD

**Issue by tambien** [chrisguttandin/standardized-audio-context#645]
Here's the reproduction only using standardized-audio-context: https://codesandbox.io/s/stupefied-surf-v8lnz

```javascript
const context = new OfflineAudioContext(2, 40, 44100);

const signal = context.createConstantSource();
signal.offset.value = 0.4;
signal.start(0);

const waveshaper = context.createWaveShaper();
waveshaper.curve = new Float32Array([-1, 0, 1]);

signal.connect(waveshaper).connect(context.destination);

context.startRendering().then(buffer => {
  console.log(buffer.getChannelData(0));
});
```

The last output sample is 0. the expected output is that it should be the same as the rest of the samples since the signal hasn't changed. 

![image](https://user-images.githubusercontent.com/1323402/63290324-12de5180-c276-11e9-92e8-f93f1101f033.png)

I've noticed a weird issue with my test case which I'm not sure if this is a codesandbox.io issue, but sometimes when i run it the last output is correct. that might just be because the page doesn't fully refresh or there's some kind of cache issue. That's why i included the screenshot. After a full page refresh, it produces the incorrect value as the last sample.

**Comment by chrisguttandin** [comment]
Thanks for reporting this. The method for checking browser support for the copy{From,To}Channel() methods was actually altering the AudioBuffer which was used for testing. This function only runs once and the result is then cached which is why you got inconsistent results.

**Comment by chrisguttandin** [comment]
It should be fixed in [v21.0.2](https://github.com/chrisguttandin/standardized-audio-context/commit/2ac868632fda91a3f8b3edf530e05c7f89bc5f48).

**Comment by tambien** [comment]
thanks for taking a look at that so quickly!
